File: module-load-balance-gateway/src/main/java/com/xmmufan/permission/domain/rbac/UserInfo.java
package com.xmmufan.permission.domain.rbac;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;
import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;
import java.time.LocalDate;

/**
 * @author 黄源钦
 * @Description
 * @Date 14:29
 */
@Entity
@Data
public class UserInfo {

    @Id
    @GeneratedValue(generator = "idGenerator")
    @GenericGenerator(name = "idGenerator", strategy = "uuid")
    @Column(length = 64, nullable = false)
    private String id;

    @OneToOne(cascade = CascadeType.REFRESH)
    @JoinColumn(name = "user_id")
    @JsonIgnore
    private User user;

    private String nickName;

    private String gender;

    private String sign;

    private LocalDate birthday;

    private String email;

    private String location;

    private String headPicture;

}

File: module-load-balance-gateway/src/main/java/com/xmmufan/permission/domain/rbac/UserInfo.kt
package com.xmmufan.permission.domain.rbac

import org.hibernate.annotations.GenericGenerator
import java.io.Serializable

import javax.persistence.*
import java.time.LocalDate

/**
 * @author 黄源钦
 * @Description
 * @Date 14:29
 */
@Entity
data class UserInfo (

    @Id
    @GeneratedValue(generator = "idGenerator")
    @GenericGenerator(name = "idGenerator", strategy = "uuid")
    @Column(length = 64, nullable = false)
    var id: String? = null,

    var nickName: String? = null,

    var gender: String? = null,

    var sign: String? = null,

    var birthday: LocalDate? = null,

    var email: String? = null,

    var location: String? = null,

    var headPicture: String? = null
): Serializable{
    companion object{
        fun empty() : UserInfo{
            return UserInfo()
        }
    }
}
